Lib/uuid.py

```diff
@@ -475,74 +475,100 @@ def _netbios_getnode():
             continue
         return int.from_bytes(bytes, 'big')
 
+def _is_darwin_greater_than_version(major_version_number):
+    """Checks if the operating system is Darwin newer than
+    the specified major version number."""
+    if sys.platform != "darwin":
+        return False
+    return (int(os.uname().release.split(".")[0]) >= major_version_number)
+
+# The uuid_generate_* functions are broken on MacOS X 10.5, as noted
+# in issue #8621 the function generates the same sequence of values
+# in the parent process and all children created using fork (unless
+# those children use exec as well).
+#
+# Assume that the uuid_generate functions are broken from 10.5 onward,
+# the test can be adjusted when a later version is fixed.
+_no_uuid_generate_time_lookup = _is_darwin_greater_than_version(9)
+_uuid_generate_time = _UuidCreate = None
+
 # Thanks to Thomas Heller for ctypes and for his help with its use here.
 
 # If ctypes is available, use it to find system routines for UUID generation.
 # XXX This makes the module non-thread-safe!
-_uuid_generate_time = _UuidCreate = None
-try:
-    import ctypes, ctypes.util
-    import sys
+def _get_uuid_generate_time():
+    """Lazy-loading of _uuid_generate_time."""
+    global _no_uuid_generate_time_lookup
```

Comment:
If I read correctly, this line is not needed.  Declaring global is only needed to assign to a name, not to read from it.

Reply:
This one is actually required. I've tried to remove it, but then the function call fails.

Reply:
Ah!  The function sets the global at line 508, that’s why.